File: game-server/src/main/java/work/fking/pangya/game/net/ServerChannelInitializer.java
package work.fking.pangya.game.net;

import io.netty.channel.Channel;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelPipeline;
import work.fking.pangya.game.GameServer;
import work.fking.pangya.networking.protocol.SimplePacketEncoder;

public class ServerChannelInitializer extends ChannelInitializer<Channel> {

    private final HelloHandler helloHandler;

    private ServerChannelInitializer(GameServer gameServer) {
        this.helloHandler = HelloHandler.create(gameServer);
    }

    public static ServerChannelInitializer create(GameServer gameServer) {
        return new ServerChannelInitializer(gameServer);
    }

    @Override
    protected void initChannel(Channel channel) {
        ChannelPipeline pipeline = channel.pipeline();

        pipeline.addLast("encoder", new SimplePacketEncoder());
        pipeline.addLast(helloHandler);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/net/ServerChannelInitializer.kt
package work.fking.pangya.game.net

import io.netty.channel.Channel
import io.netty.channel.ChannelInitializer
import work.fking.pangya.game.GameServer
import work.fking.pangya.networking.protocol.SimplePacketEncoder

class ServerChannelInitializer(gameServer: GameServer) : ChannelInitializer<Channel>() {
    private val helloHandler: HelloHandler

    init {
        helloHandler = HelloHandler(gameServer)
    }

    override fun initChannel(channel: Channel) {
        val pipeline = channel.pipeline()
        pipeline.addLast("encoder", SimplePacketEncoder())
        pipeline.addLast(helloHandler)
    }
}
